cachito/web/content_manifest.py

```diff
@@ -85,9 +104,50 @@ def process_go_package(self, package, dependency):
         :param Dependency dependency: the go-package package dependency to process
         """
         if dependency.type == "go-package":
-            icm_dependency = {"purl": to_purl(dependency)}
+            if dependency.version and dependency.version.startswith("."):
+                dep_purl = self._get_local_go_package_dep_purl(package, dependency)
+            else:
+                dep_purl = to_purl(dependency)
+
+            icm_dependency = {"purl": dep_purl}
             self._gopkg_data[package]["dependencies"].append(icm_dependency)
 
+    def _get_local_go_package_dep_purl(self, package: "Package", dependency: "Package") -> str:
+        """
+        Get the purl for a local go-package dependency.
+
+        :param Package package: the go-package package to process
+        :param Dependency dependency: the go-package package dependency to process
+        """
+        if not dependency.version or not dependency.version.startswith("."):
+            raise ValueError("{dependency} has an invalid version for a local dependency")
+
+        modules = {module.name: module for module in self.packages if module.type == "gomod"}
```

Comment:
Minor nitpick: I would prefer if we didn't have to re-create this dict every time. Maybe something like a `go_modules_by_name` [cached_property](https://docs.python.org/3/library/functools.html#functools.cached_property) could work?

Reply:
Added :+1: